rpc/client/mock/abci.go

```diff
@@ -0,0 +1,186 @@
+package mock
+
+import (
+	abci "github.com/tendermint/abci/types"
+	"github.com/tendermint/tendermint/rpc/client"
+	ctypes "github.com/tendermint/tendermint/rpc/core/types"
+	"github.com/tendermint/tendermint/types"
+)
+
+// ABCIApp will send all abci related request to the named app,
+// so you can test app behavior from a client without needing
+// an entire tendermint node
+type ABCIApp struct {
+	App abci.Application
+}
+
+func (a ABCIApp) _assertABCIClient() client.ABCIClient {
```

Comment:
Not needed by nice helpers. Talked with Bucky about them, he liked the idea too.

Basically, when you create Struct explicitly to fulfill an interface, you can assert so if the methods don't match, you get a compile error in the module itself, not when someone else tried to use it.

Matt had this problem with endblock changing SIG and then quietly being ignored in basecoin without any compiler warnings.